# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: tools/profiler/core/ProfiledThreadData.cpp
# Commit: 524f84e0fc78
# Full Hash: 524f84e0fc78eb8689dc6782f16f5bf71e4e5edf
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2019-09-18 09:56:40
# Regressor Bug: 1576551
# File Overlap Count: 2
# Description:
#   Bug 1576551 - Use BlocksRingBuffer in ProfileBuffer - r=gregtatum
#   
#   This just replaces `ProfileBuffer`'s self-managed circular buffer with a
#   `BlocksRingBuffer`.
#   
# ==============================================================================

diff -r af093f06fb74 -r 524f84e0fc78 tools/profiler/core/ProfiledThreadData.cpp
--- a/tools/profiler/core/ProfiledThreadData.cpp	Wed Sep 18 01:19:10 2019 +0000
+++ b/tools/profiler/core/ProfiledThreadData.cpp	Wed Sep 18 01:19:12 2019 +0000
@@ -38,7 +38,8 @@
                                     double aSinceTime, bool JSTracerEnabled,
                                     ProfilerCodeAddressService* aService) {
   if (mJITFrameInfoForPreviousJSContexts &&
-      mJITFrameInfoForPreviousJSContexts->HasExpired(aBuffer.mRangeStart)) {
+      mJITFrameInfoForPreviousJSContexts->HasExpired(
+          aBuffer.BufferRangeStart())) {
     mJITFrameInfoForPreviousJSContexts = nullptr;
   }
 
@@ -287,7 +288,8 @@
   MOZ_RELEASE_ASSERT(aContext);
 
   if (mJITFrameInfoForPreviousJSContexts &&
-      mJITFrameInfoForPreviousJSContexts->HasExpired(aBuffer.mRangeStart)) {
+      mJITFrameInfoForPreviousJSContexts->HasExpired(
+          aBuffer.BufferRangeStart())) {
     mJITFrameInfoForPreviousJSContexts = nullptr;
   }
 